src/AggregateFunctions/AggregateFunctionSequenceNextNode.cpp

```diff
@@ -0,0 +1,114 @@
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+#include <AggregateFunctions/AggregateFunctionSequenceNextNode.h>
+#include <AggregateFunctions/Helpers.h>
+#include <AggregateFunctions/FactoryHelpers.h>
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <ext/range.h>
+#include "registerAggregateFunctions.h"
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int BAD_ARGUMENTS;
+}
+
+namespace
+{
+
+template <typename TYPE>
+inline AggregateFunctionPtr createAggregateFunctionSequenceNodeImpl(const DataTypePtr data_type, const DataTypes & argument_types, bool descending_order)
+{
+    if (argument_types.size() == 2)
+    {
+        // If the number of arguments of sequenceNextNode is 2, the sequenceNextNode acts as sequenceFirstNode.
+        if (descending_order)
+            return std::make_shared<SequenceFirstNodeImpl<TYPE, NodeString, true>>(data_type);
+        else
+            return std::make_shared<SequenceFirstNodeImpl<TYPE, NodeString, false>>(data_type);
+    }
+    else
+    {
+        if (descending_order)
+            return std::make_shared<SequenceNextNodeImpl<TYPE, NodeString, true>>(data_type, argument_types);
+        else
+            return std::make_shared<SequenceNextNodeImpl<TYPE, NodeString, false>>(data_type, argument_types);
+    }
+}
+
+template <UInt64 MaxArgs>
+AggregateFunctionPtr createAggregateFunctionSequenceNode(const std::string & name, const DataTypes & argument_types, const Array & parameters)
+{
+    bool descending_order = false;
+
+    if (parameters.size() == 1)
+    {
+        auto type = parameters[0].getType();
+        if (type != Field::Types::Int64 && type != Field::Types::UInt64)
+               throw Exception("The first parameter for aggregate function " + name + " should be 0 or 1", ErrorCodes::BAD_ARGUMENTS);
+
+        descending_order = parameters[0].get<UInt64>();
+    }
+    else
+        throw Exception("Incorrect number of parameters for aggregate function " + name + ", should be 1",
+            ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+    if (argument_types.size() < 2)
+        throw Exception("Aggregate function " + name + " requires at least two arguments.", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+    else if (argument_types.size() > MaxArgs)
+        throw Exception("Aggregate function " + name + " requires at most 34(timestamp, value_column, 31 events) arguments.", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+    for (const auto i : ext::range(2, argument_types.size()))
+    {
+        const auto * cond_arg = argument_types[i].get();
+        if (!isUInt8(cond_arg))
```

Comment:
In some cases,  event chains refer to another columns.
For example, 
```
dt                                    id     action            page     
2021-02-12 00:00:01     1     click_btn1      main
2021-02-12 00:01:01     1     click_btn2      shopping
2021-02-12 00:02:01     1     click_btn3      news
2021-02-12 00:03:01     1     click_btn4      sports
```
```
SELECT sequenceNextNode(1)(dt, page, action = 'click_btn1', action = 'click_btn2', page = 'news') AS next_node  FROM test_table;
```
So that I created column with result of comparison.